src/Plugin/AddPathPlugin.php

```diff
@@ -42,18 +35,33 @@ public function __construct(UriInterface $uri)
     }
 
     /**
+     * Adds a prefix in the beginning of the URL's path. We will not add the prefix
+     * if it already exists. This will of course not cover all the edge cases. For
+     * example when you have a valid URL like `https://example.com/api/api/foo` and
+     * you are using the prefix "/api".
+     *
+     * Some of the other edge cases are:
+     *  - request objects can be reused
+     *  - plugin chains could be restarted
+     *  - multiple AddPathPlugins may exist in one chain
+     *
+     * If one would like to cover all these scenarios one would have to add a custom
+     * header to the request to indicate it has been rewritten by the plugin. This
+     * implementation does not aim for pleasing all edge cases rather to be great
+     * value for the 95% other use cases.
+     *
```

Comment:
thanks! i would prefer to be a bit more explicit:

```
    /**
     * Adds a prefix in the beginning of the URL's path.
     *
     * The prefix is not added if that prefix is already on the URL's path. This will fail on the edge
     * case of the prefix being repeated, for example if `https://example.com/api/api/foo` is a valid
     * URL on the server and the configured prefix is `/api`.
     *
     * We looked at other solutions, but they are all much more complicated, while still having edge
     * cases:
     *  - Doing an spl_object_hash on `$first` will lead to collisions over time because over time the
     *    hash can collide.
     * - Have the PluginClient provide a magic header to identify the request chain and only apply
     *   this plugin once.
     *
     * There are 2 reasons for the AddPathPlugin to be executed twice on the same request:
     * - A plugin can restart the chain by calling `$first`, e.g. redirect
     * - A plugin can call `$next` more than once, e.g. retry
     *
     * Depending on the scenario, the path should or should not be added. E.g. `$first` could
     * be called after a redirect response from the server. The server likely already has the
     * correct path.
     *
     * No solution fits all use cases. This implementation will work fine for the common use cases.
     * If you have a specific situation where this is not the right thing, you can build a custom plugin
     * that does exactly what you need.
     */